langchain4j-azure-open-ai/src/main/java/dev/langchain4j/model/azure/InternalAzureOpenAiHelper.java

```diff
@@ -261,7 +226,7 @@ private static BinaryData getParameters(ToolSpecification toolSpecification) {
         if (toolSpecification.parameters() != null) {
             return toOpenAiParameters(toolSpecification.parameters());
         } else {
-            return toOpenAiParametersOld(toolSpecification.toolParameters());
+            return toOpenAiParametersOld(toolSpecification.parameters());
```

Comment:
This change seems wrong and should not be required. As you can see, method name is toOpenAiParameters**_Old_** and is supposed to convert deprecated `ToolParameters` (that some users might still use in their code).
Handling for new parameters happens 2 lines above. `toOpenAiParametersOld` should stay as-is and convert `ToolParameters`